lib/mongo/protocol/utils.ex

```diff
@@ -15,8 +15,14 @@ defmodule Mongo.Protocol.Utils do
   end
 
   def command(id, command, s) do
-    op = op_query(coll: namespace("$cmd", s, nil), query: BSON.Encoder.document(command),
+    op = case command do
+      [authenticate: 1, user: _username, mechanism: "MONGODB-X509"] ->
+        op_query(coll: namespace("$cmd", nil, "$external"), query: BSON.Encoder.document(command),
                   select: "", num_skip: 0, num_return: 1, flags: [])
+      _command ->
+        op_query(coll: namespace("$cmd", s, nil), query: BSON.Encoder.document(command),
+                  select: "", num_skip: 0, num_return: 1, flags: [])
+    end
```

Comment:
This could probably be cleaned up to remove the duplication. To make this less brittle, you may want to use `Keyword.get/2` to grab only the important parts of the X509 message.

Reply:
I can definitely extract out the namespace difference to reduce duplication. I'll try setting the namespace with `Keyword.get/2` but I'm not sure if it's less brittle. I think the case looks a little better, but I'll update the PR and you can decide.